keras_nlp/utils/text_generation.py

```diff
@@ -502,23 +503,30 @@ def token_probability_fn(inputs):
             "tf.function or run `tf.config.run_functions_eagerly(True)` to run "
             "tf.function in eager mode."
         )
-    if k <= 0:
-        raise ValueError(f"`k` should be strictly positive. Received: `k={k}`.")
 
     prompt = validate_prompt(prompt)
     input_is_1d = prompt.shape.rank == 1
     if input_is_1d:
         prompt = prompt[tf.newaxis, :]
     validate_token_probability_fn(token_probability_fn, prompt)
 
+    if k <= 0:
+        raise ValueError(f"`k` should be strictly positive. Received: `k={k}`.")
+    # If k is greater than the vocabulary size, use the entire vocabulary.
+    pred = token_probability_fn(prompt)
+    if k > pred.shape[1]:
+        logging.warning(
+            f"`k`={k} larger than vocabulary size of {pred.shape[1]}. "
```

Comment:
You could extend the backticks to the whole `k={k}` expression, similar to the value error above. Shows that the whole statement is valid python. Same on next line.

Also start the warning with "Received `k={k}`...". We use that as a common keyword to indicate user input.